backstory/src/main/java/com/google/sps/vision/VisionManagerImpl.java

```diff
@@ -0,0 +1,176 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.vision;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.protobuf.ByteString;
+import java.io.IOException;
+import java.lang.IllegalArgumentException;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class VisionManagerImpl implements VisionManager {
+  private final byte[] rawImageData;
+  private final List<EntityAnnotation> labelAnnotations;
+
+  /**
+   * VisionManager Manages the gathering and packaging of Vision API image analytics.
+   * A VisionManager object represents an image annotated with analytics from Vision API.
+   *
+   * The image is represented as the byte data in the rawImageData byte array, and the annotated
+   * analytics are represented as labelAnnotations, a list of Vision EntityAnnotation objects, which
+   * each represent individual labels.
+   *
+   * Abstraction Function(rawImageData, labelAnnotations) =
+   * An image represented by the bytes in rawImageData, with a list of labels annotated
+   * to the image, represented by the EntityAnnotation(s) in labelAnnotations.
+   *
+   * Representation Invariant:
+   * rawImageData will always represent the same image, it is immutable. It must be non-null. It
+   * must be non-empty. Once set, labelAnnotations will never change, it is immutable. It must be
+   * non-null.
+   *
+   * Safety from Representation Exposure:
+   * All representation fields are immutable.
+   */
+
+  /** Asserts the rep invariants */
+  private void checkRep() {
```

Comment:
Even though this is private, a more descriptive method name should be utilized here, especially because Representation Invariant is not a straight forward keyword - consider expanding on it in the javadoc for the method.

Reply:
Sounds good - I added it to the javadoc in our case.